test/lib/utils.public.test.js

```diff
@@ -184,72 +196,85 @@ tap.test('prettifyObject', t => {
 })
 
 tap.test('prettifyTime', t => {
-  const { prettifyTime } = utils
-
   t.test('returns `undefined` if `time` or `timestamp` not in log', async t => {
-    const str = prettifyTime({ log: {} })
-    t.is(str, undefined)
+    const prettified = {}
+    prettifyTime({}, { prettified })
+    t.is(prettified.prettifiedTime, undefined)
```

Comment:
I might have missed why we are modifying an object instead of returning a string. Would you mind to articulate?

Reply:
The parse process is essentially a pipeline, in which the structure of the input and output of each function in the pipeline is identical. So a string can't be returned directly because it would be fed into the next parser and we'd lose the log object.

There are exceptions to this. The `primitives` parser returns a primitive value, but then it short-circuits the log processor, bypassing additional parsers and builders.

The input and output of the `parse` function is the log entry which is eventually fed to the builders, though it may be modified by the parsers as necessary. It seemed to make more sense to "cache" the prettified values on the `context.prettified` object because it mirrors the behavior that already existed before my changes and it also makes the API clearer by not "dirtying" the log object.